### DISCUSSION TITLE
Include CSS in dist

### DISCUSSION THREAD

**Discussion by cgatian** [chrisweb/rehype-github-alerts#11] [Ideas]
Would it be possible to ship the CSS stylesheet as a part of the package?

I can copy-paste it myself, but it might be nice to include it in the bundle.

**Comment by chrisweb** [comment]
@cgatian we had a similar discussion not so long where I talked about the pro and cons I see in adding a CSS to the package itself: https://github.com/chrisweb/rehype-github-alerts/discussions/9#discussioncomment-11284610

So tldr I'm not convinced this would be an improvement

If you have to copy paste the CSS from a Readme that has quite good visibility (gets seen easily by people wanting to use the package) versus having the CSS in a folder in the package itself (where it might get overlooked and would probably need to get mentioned in the readme anyway), is that really better? I mean a CSS file would not get imported, it would just be there in some folder and then the dev would need to copy paste the CSS from that file to his own CSS (or copy and eventually transform it into one of the many options you have when it comes to applying CSS / styling in a Nextjs/Reatc project)

**Comment by cgatian** [comment]
Yeah, I was just thinking that if the structure of the DOM this plugin outputs changes, it might be nice to keep the styles in lockstep.

Frameworks sometimes [rely on a setup](https://panda-css.com/docs/installation/nextjs#configure-the-entry-css-with-layers) step to include their CSS within your page, so I don't think this is asking for something out of the ordinary. I'm not here to try to convince you; I think what you have works. This is just a suggestion.

**Reply by chrisweb** [reply]
@cgatian I moved this to discussion, so if you or anybody else wants to discuss this further we can do so here

First I quickly wanted to clarify that I said that I was not convinced, but I didn't imply that you have to convince me, in general if I have the time to do so I will answer questions, discuss ideas as much as I can, after all isn't opensource about NOT expecting someone to do something but be glad when they do ;)

But to the issue itself, we both agree the current solution is "viable" (having a CSS template in the readme), so at least there is no urgency to act and we have plenty of time to discuss and eventually find an ideal solution some day

I also agree with you that it is a good idea to look at other frameworks, when I read your post it made me wonder how does [remark-gfm](https://github.com/remarkjs/remark-gfm) deal with this? did they discuss this in the past (I had no time to check it out, yet)

**Reply by chrisweb** [reply]
I thought about it briefly yesterday, for me this plugin should produce a github-like HTML that has github-like classes, for the CSS itself (due to the blooming landscape of frameworks, CSS libraries, UI kits, post processing tools, ...) I have the feeling there is no way to ship CSS in a way that makes it a no brainer for devs, it would require a solution that works with (can be imported / included) in every stack imaginable

So maybe this package shouldn't try too hard to include a universal styling solution and instead we could have smaller packages, that get maintained seperatly, each packages would style the github alerts by using features that are specific to the tool that it got build for, for example you could have a "remark-github-alerts-css-modules" package that has a css file that css modules can import, then another package would offer a solution that works with tailwind, or styleX, or MUI (pigment css), panda css...
